crates/telio-relay/src/derp/mod.rs

```diff
@@ -115,6 +120,10 @@ struct State {
     derp_poll_session: Session,
     /// Cache the result of derp polling
     remote_peers_states: PeersStatesMap,
+    /// Connectivity data aggregator
+    aggregator: Option<Arc<ConnectivityDataAggregator>>,
```

Comment:
I think it would be better to extract an interface out of `ConnectivityDataAggregator` and provide a no-op implementation for it, so that we can avoid checking such optionals each time new derp/nat event is to be reported.

Reply:
I guess I can just use the flags which are already present flag for that?

Reply:
Maybe, I'm not exactly sure what you have in mind. But overall I think it's probably better to do one conditional and either create a real aggregator or a no-op one and than use it unconditionally everywhere (derp, nat, nurse reporting) instead of using optional/flags in all those places each time you need to call a method on the aggregator.

Reply:
Hmm, I don't like that idea too much tbh, sounds like an overengineering to me

Reply:
In this case, I think we have only two call-sites call into aggregator from DERP module. So the "saving" on if's are not super big. But I I do agree, that lower number of conditionals makes things simpler. Additionally exported trait allows for easier mock implementations in unit tests.

Reply:
Oh, but aggregator is already mocked in unit tests. so nvm, about mocking.